core_lib_generator/file_generators/test_generator.py

```diff
@@ -0,0 +1,60 @@
+from core_lib.data_transform.helpers import get_dict_attr
+from core_lib.helpers.string import snake_to_camel, camel_to_snake
+from core_lib_generator.file_generators.template_generator import TemplateGenerator
+from core_lib_generator.generator_utils.formatting_utils import add_tab_spaces, remove_line
+
+
+class TestGenerateTemplate(TemplateGenerator):
+    def generate(self, template_content: str, yaml_data: dict, core_lib_name: str, file_name: str) -> str:
+        updated_file = template_content
+        camel_core_lib = snake_to_camel(core_lib_name)
+        core_lib_import = f'from {core_lib_name}.{core_lib_name} import {camel_core_lib}'
+        updated_file = updated_file.replace('# template_core_lib_import', core_lib_import)
+        updated_file = updated_file.replace('TemplateCoreLib', camel_core_lib)
+        updated_file = updated_file.replace('template_snake_core_lib', core_lib_name)
+        if get_dict_attr(yaml_data, 'services'):
+            service_imports_list = []
+            service_test_functions = []
+            for service in get_dict_attr(yaml_data, 'services'):
+                test_function = []

```

Comment:
no code duplication!!!
this loop is the same as the dataaccess loop. create a function and pas parameters